OSDOCS#9282: Creating certificates using an issuer for cert-manager (#70018)
openshift/openshift-docs: modules/cert-manager-certificate-ingress.adoc

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-creating-certificate.adoc

:_mod-docs-content-type: PROCEDURE
[id="cert-manager-certificate-ingress_{context}"]
= Creating certificates for Ingress Controller

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* You have installed {cert-manager-operator} 1.13.0 or above.

.Procedure

. Create an issuer. For more information, see "Configuring an issuer" in "Additional Resources" section.

. Create a certificate:

.. Create a YAML file, for example, `certificate.yaml`, that defines the `Certificate` object:
+
.Example `certificate.yaml` file
+
[source, yaml]
----
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: <tls_cert> #<1>
  namespace: openshift-ingress
spec:
  isCA: false
  commonName: "<common_name>" #<2>
  secretName: <tls_cert> #<3>
  dnsNames:

>   - "*.new-ingress.<sub.domai.ns-domain-x.y.z.com>" #<4>

Reviewer: This has to be:
```
dnsNames:
- apps.<cluster_domain_name>
- *.apps.<cluster_domain_name>
```
and commonName field has to be exact same as the first one i.e.
```
commonName: apps.<cluster_domain_name>
```

Author: do you think the corresponding callouts need any changes or does the existing description work?

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-creating-certificate.adoc

:_mod-docs-content-type: PROCEDURE
[id="cert-manager-certificate-ingress_{context}"]

> = Creating certificates for Ingress Controller

Reviewer 2: ```suggestion
= Creating certificates for the Ingress Controller
```

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.

> * You have installed {cert-manager-operator} 1.13.0 or above.

Reviewer 2: Same comment about "the":
```suggestion
* You have installed the {cert-manager-operator} 1.13.0 or above.
```

> * You have installed {cert-manager-operator} 1.13.0 or above.

Reviewer 2: Also same comment about "or later":
```suggestion
* You have installed {cert-manager-operator} 1.13.0 or later.
```

.Procedure

> . Create an issuer. For more information, see "Configuring an issuer" in "Additional Resources" section.

Reviewer 2: Same comment about this maybe being more of a prereq than a first step, up to you.

. Create a certificate:

> .. Create a YAML file, for example, `certificate.yaml`, that defines the `Certificate` object:

Reviewer 2: ```suggestion
.. Create a YAML file, for example `certificate.yaml`, that defines the `Certificate` object:
```

Author: same response as above

+
.Example `certificate.yaml` file

> +

Reviewer 2: I think this line can be deleted

[source, yaml]
----
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: <tls_cert> #<1>
  namespace: openshift-ingress
spec:
  isCA: false
  commonName: "apps.<cluster_base_domain>" #<2>

>   secretName: <tls_cert> #<3>

Reviewer 2: Same comment and suggestion as I made in the other module:
```suggestion
  secretName: <secret_name> #<3>
```

  dnsNames:
  - "apps.<cluster_base_domain>" #<4>
  - "*.apps.<cluster_base_domain>" #<4>
  issuerRef:
    name: <issuer_name> #<5>
    kind: Issuer
----
<1> Provide a name for the certificate.
<2> Specify the common name (CN).
<3> Specify the name of the secret to create that contains the certificate.
<4> Specify the DNS name of the ingress.
<5> Specify the name of the issuer.

.. Create the `Certificate` object by running the following command:
+
[source, terminal]
----
$ oc create -f certificate.yaml
----

> . Replace the default ingress certificate. For more information, see "Replacing the default ingress certificate" section in "Additional resources".

Reviewer 2: ```suggestion
. Replace the default ingress certificate. For more information, see "Replacing the default ingress certificate" in the "Additional resources" section.
```

.Verification

* Verify that the certificate is created and ready to use by running the following command:
+ 
[source, terminal]
----
$ oc get certificate -w -n openshift-ingress
----
+

> Once certificate is in `Ready` status, Ingress Controller on your cluster can start using the generated certificate secret.

Reviewer 2: ```suggestion
Once the certificate status is `Ready`, the Ingress Controller on your cluster can start using the generated certificate secret.
```

* You have installed {cert-manager-operator}.

.Procedure

. Create an issuer. For more information, see "Configuring an issuer" in "Additional Resources" section.

. Create a certificate:

.. Create a YAML file, for example, `certificate.yaml`, that defines the `Certificate` object:
+
.Example `certificate.yaml` file
+
[source, yaml]
----
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: <tls_cert> #<1>
  namespace: openshift-ingress
spec:
  isCA: false
  commonName: "<common_name>" #<2>
  secretName: <tls_cert> #<3>
  dnsNames:
  - "*.new-ingress.<sub.domai.ns-domain-x.y.z.com>" #<4>
  issuerRef:
    name: <issuer_name> #<5>
    kind: Issuer
----
<1> Provide a name for the certificate.
<2> Specify the common name (CN).
<3> Specify the name of the secret to create that contains the certificate.
<4> Specify the DNS name of the ingress.
<5> Specify the name of the issuer.

.. Create the `Certificate` object by running the following command:
+
[source, terminal]
----
$ oc create -f certificate.yaml
----

. Replace the default ingress certificate. For more information, see "Replacing the default ingress certificate" section in "Additional resources".
+

> once certificate is ready, application workloads on your cluster can start using the generated certificate secret.

Reviewer: I am hoping this is a typo, as same sentence is repeated after `Verification` step.

Author: yes, that's a typo. Fixed.

.Verification

* Verify that the certificate is created and ready to use by running the following command:
+ 
[source, terminal]
----
$ oc get certificate -w -n openshift-ingress
----
+

> Once certificate is in `Ready` status, workloads on your cluster can start using the generated certificate secret.

Reviewer: ```suggestion
Once certificate is in `Ready` status, ingress controller on your cluster can start using the generated certificate secret.
```

Author: Done
